fix added for line numbering using lineno package

johnjarmitage opened the pull request:
The original Nature Springer template used a lazy method for line numbering that made no sense. I have replaced it with a line numbering using the `lineno` package and [this](https://tex.stackexchange.com/questions/461186/how-to-use-lineno-with-amsmath-align) fix so that `amsmath` equations have their lines numbered too.

I also added the `sn-basic.bst` to the `tempate.yml` file.

I have tested it only on one tex file but it looks like it works.

johnjarmitage committed:
fix added for line numbering usinf lineno package with equation lines numbered

dressedfez commented:
Can you elaborate on what improvement your change brings? I need to test it as well. The current implementation used the approach provided by the Springer documentation as referred to in the README.

johnjarmitage commented:
I switched the `options.line_numbers` from using the Springer line numbering method in their `sn-jnl.cls` file to activating the `\linenumbers` option for the `lineno` package (see line 144 of `template.tex`). So that this works with equatios I also added the fix that I referenced in the template file. THe improvement is that the lines are numberd, before this they were not, instead there was simply a vertical "ruler" of numbers.

dressedfez commented:
Great. Sounds like a legit improvement. I will test it and hope to merge it during the weekend. @rowanc1 @stevejpurves any further comments?

rowanc1 commented:
This is awesome! Thank you both!
Really cool to see the improvements to the templates. 🚀

dressedfez commented:
Thanks for the contribution. I especially like that the line numbering is now unique for a two column paper. I will merge the PR. My tests have been successful.
